### DISCUSSION TITLE
Is there any way to deep sorting with content api ?

### DISCUSSION THREAD

**Discussion by ninaninani** [strapi/strapi#7315] [Legacy Topics]
Hello, I have a collection with this structure:

![image](https://user-images.githubusercontent.com/5014071/89300693-2bb86080-d69b-11ea-9d0e-d8ae1a54bfd2.png)

I need to sort the collection with "published_date" field which is in "header".
I tried GET /posts?_sort=header.date:DESC
It doesn't show any error but it doesn't sort correctly either.

Is there any way I can sort it correctly with the current filed structure ?

**Comment by derrickmehaffy** [comment]
Hello, currently this option doesn't exist for deep sorting (only deep filtering). Can you please open up a [feature request](https://github.com/strapi/strapi/issues/new?template=FEATURE_REQUEST.md) and provide as much information as you can about your use-case as you can. 

For a workaround for now, you will need to build a custom mongoose query. Please see this documentation: https://strapi.io/documentation/v3.x/concepts/queries.html#custom-queries
